spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkDaysFunction.java

```diff
@@ -39,6 +39,9 @@ public void testDates() {
     assertThat(scalarSql("SELECT system.days(date('2017-12-01'))"))
         .as("Expected to produce 2017-12-01")
         .isEqualTo(Date.valueOf("2017-12-01"));
+    assertThat(scalarSql("SELECT system.day(date('2017-12-01'))"))
```

Comment:
Can we also update negative test cases? e.g. testWrongNumberOfArguments 
The function name is hard-coded in each function class in `name()` and `canonicalName()` methods. We may want to add a constructor taking the function name to report the correct function name.

Reply:
I didn't repeat all the test cases, as it does not seem necessary. I just added a sample of each (posititive) case with the singular form to demonstrate that the form can be used.

Reply:
It hides the fact that these functions can throw different function names which users specified.

Reply:
Ok, I see what you're getting at.
I decided to add constructors to the function so that we know what form (singular or plural) is being used; singular is the default. I have accordingly documented (in javadoc) the singular form as the first example with the plural form as the alternate. I didn't change the names of the classes as that would be too much work and not worth it.